### PR TITLE
Add conformance to SignalProducerConvertible for Result

### PR DESCRIPTION
**Pull request by Qata**
#### Checklist
- [X] Updated CHANGELOG.md.
This implements the suggestion in #409

### CODE REVIEW

**Comment by NachoSoto** [comment]
That's a neat idea! Any chance you can add a few tests for this?

**Comment by Qata** [comment]
I didn't because there isn't any current conformance tests at all, but I can.

File: Sources/ResultExtensions.swift [hunk 1]
```diff
@@ -11,3 +11,15 @@ import Result
 internal func materialize<T>(_ f: () throws -> T) -> Result<T, AnyError> {
 	return materialize(try f())
 }
+
+extension Result: SignalProducerConvertible {
+	public var producer: SignalProducer<Value, Error> {
+		switch self {
```
Comment (ikesyo) [inline Sources/ResultExtensions.swift]: You can use the initializer here: https://github.com/ReactiveCocoa/ReactiveSwift/blob/55ac251953f0d0fe0076d7b7a5b9a3c2b9c382bb/Sources/SignalProducer.swift#L156

Comment (Qata) [reply Sources/ResultExtensions.swift]: Done 👍

**Comment by Qata** [comment]
@NachoSoto Tests added.

File: Tests/ReactiveSwiftTests/SignalProducerSpec.swift [hunk 2]
```diff
@@ -484,6 +484,34 @@ class SignalProducerSpec: QuickSpec {
 				expect(error) == operationError
 			}
 		}
+		
+		describe("init(_:) base") {
```
Comment (NachoSoto) [inline Tests/ReactiveSwiftTests/SignalProducerSpec.swift]: This should be `Result.producer`, right?

File: Tests/ReactiveSwiftTests/SignalProducerSpec.swift [hunk 3]
```diff
@@ -484,6 +484,34 @@ class SignalProducerSpec: QuickSpec {
...
+				let operationReturnValue = "OperationValue"
+				
+				let signalProducer = SignalProducer<String, AnyError>(Result.success(operationReturnValue))
+				
+				var error: Error?
+				signalProducer.startWithFailed {
+					error = $0
+				}
+				
+				expect(error).to(beNil())
```
Comment (NachoSoto) [inline Tests/ReactiveSwiftTests/SignalProducerSpec.swift]: You should check that the value was sent, and that it completed.

File: Tests/ReactiveSwiftTests/SignalProducerSpec.swift (hunk 2 above)
Comment (Qata) [reply Tests/ReactiveSwiftTests/SignalProducerSpec.swift]: Well technically I'm testing `SignalProducer.init(_ base:)` as well but I can call it `Result.producer`.

**Comment by Qata** [comment]
@NachoSoto Amended tests to address comments.

**Comment by andersio** [comment]
Resolves #409.

**Comment by Qata** [comment]
@andersio Amended

**Pull request merged**